src/Uno.Core.Tests/Collections/ObservableCollectionExtensionsFixture.cs

```diff
@@ -267,5 +268,68 @@ public void RemoveSingle_WithResults()
 			Assert.AreEqual(0, results.Added.Count());
 			Assert.AreEqual(0, results.Moved.Count());
 		}
+
+		[TestMethod]
+		public void Update_KeyEquatable()
```

Comment:
What happens if the same item is present multiple times? or if an two different items end being equal ?

Reply:
Do you mean multiple times in the old version of the collection (or the new)? It shouldn't really matter, if both old and new sequences are sequence-key-equal (ie if there are duplicates in the same locations) it will replace the ones with a newer version. If they're not key-equal it will use the old code path.

"if an two different items end being equal" not sure what this means, if two items return equal (not key-equal) then it will keep the old item, in the new code path as in the existing path. Obviously it's on the consumer to define equality correctly.